### PR TITLE
Update docs for proto toolchain

### PR DESCRIPTION
**Pull request by liucijus**
Docs updates related to a new `scala_proto_deps_toolchain` toolchain

### CODE REVIEW

File: docs/scala_proto_library.md [hunk 1]
```diff
@@ -32,25 +32,59 @@ generated by the [ScalaPB compiler](https://github.com/scalapb/ScalaPB).
...
+
+| Attribute name                | Description                                           |
+| ----------------------------- | ----------------------------------------------------- |
+| name                          | `Name, required` <br> A unique name for this toolchain.
+| with_grpc                     | `boolean, optional (default False)` <br> Enables generation of grpc service bindings for services.
+| with_flat_package             | `boolean, optional (default False)` <br> When true, ScalaPB will not append the protofile base name to the package name.
+| with_single_line_to_string    | `boolean, optional (default False)` <br> Enables generation of toString() methods that use a single line format.
+| blacklisted_protos            | `List of labels, optional` <br> List of protobuf targets to exclude from recursive building.
+| code_generator                | `Label, optional (has default)` <br> Which code generator to use. A sensible default is provided.
+| named_generators              | `String dict, optional` <br>
```
Comment (simuons) [inline docs/scala_proto_library.md:65]: Maybe add explanation what this dict contains. ie a mapping from protoc plugin name to a class name implementing `protocbridge.ProtocCodeGenerator`

File: docs/scala_proto_library.md [hunk 2]
```diff
@@ -32,25 +32,59 @@ generated by the [ScalaPB compiler](https://github.com/scalapb/ScalaPB).
...
+| extra_generator_dependencies  | `List of labels, optional` <br>
```
Comment (simuons) [inline docs/scala_proto_library.md:66]: Maybe add explanation that this contains a list of jars where named generators are loaded from.

File: docs/scala_proto_library.md (hunk 1 above)
Comment (liucijus) [reply docs/scala_proto_library.md:65]: This table is from previous doc, I would like to keep changes not related to deps toolchain for another PR

**Review by simuons — APPROVED** [review]

File: docs/scala_proto_library.md [hunk 3]
```diff
@@ -32,25 +32,59 @@ generated by the [ScalaPB compiler](https://github.com/scalapb/ScalaPB).
...
+| scalac                        | `Label, optional (has default)` <br> Target for scalac. A sensible default is provided.
+
+To configure dependencies, configure a different `scala_proto_deps_toolchain` and declare it in a `BUILD` file:
```
Comment (ittaiz) [inline docs/scala_proto_library.md:69]: Maybe "to configure custom dependencies"? Or does somewhere else clarify that by default you don't need to do anything

Comment (liucijus) [reply docs/scala_proto_library.md:72]: I updated docs with clarification

**Review by ittaiz — APPROVED** [review]

**Pull request merged**